### DISCUSSION TITLE
Z_Calibrate Panel

### DISCUSSION THREAD

**Discussion by RitzSpecial** [KlipperScreen/KlipperScreen#1283] [Q&A] [answered]
I can run Z_ENDSTOP_CALIBRATE from the command prompt but when I try to use the KlipperScreen Panel it doesn’t want to do endstop calibration. All the buttons except start and pos are inactive. Pressing start swaps between endstop and probe. The gtk code is a little deep and I can’t understand why the presses on start cause the swap unless that is what is meant by asking the user which one? I can’t find anyone on Discord using the panel and don’t mind debugging but I need a little help.

**Comment by alfrix** [comment]
if there is more than 1 method it should ask you like this
![imagen](https://github.com/KlipperScreen/KlipperScreen/assets/1247237/d5b1f297-4570-4017-add5-97816e3b9b2c)
i have noticed that it looks terrible, i remember that it looked good when it was first introduced, but changes in the themes may have created this, i will fix the styling

**Comment by alfrix** [comment]
after some restyling
![imagen](https://github.com/KlipperScreen/KlipperScreen/assets/1247237/331e84fd-788e-4723-9609-ae6ddc95c812)

**Comment by RitzSpecial** [comment]
That will be very helpful. I can’t wait to get a chance to try it. I thought I had tried pressing all around that area but now I’ll be able to understand what’s going on.

**Comment by RitzSpecial** [comment, marked as answer]
Seeing the new layout helped me understand. That was exactly what I needed! Printed the first object today and it came out very nice. Now I can continue with the tuning!
